pkg/inject/inject_test.go

```diff
@@ -713,3 +713,167 @@ func Test_escapeJsonPointer(t *testing.T) {
 		})
 	}
 }
+
+func Test_removeContainerVolume(t *testing.T) {
```

Comment:
the unit test is great but we must have integration tests that will reproduce the conflict and will prove that this method is actually invoked correctly and the patches are doing their job in transformation. to prove it, we can use the same validation method kubernetes api is using to make sure we are passing the validation ([ValidateVolumeMounts](https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/validation/validation.go#L2595)). also, we need to check it in both injection strategies. 
can you please add those tests to `transform_test.go`?